### PR TITLE
fix(healthcare): Wait for dataset operation to complete before continuing with tests

### PR DESCRIPTION
**Pull request by nikklassen**
This is basically the same as the fix in the Python tests GoogleCloudPlatform/python-docs-samples#7501

### CODE REVIEW

File: healthcare/dicom_test.go [hunk 1]
```diff
@@ -58,9 +59,13 @@ func TestDICOMStore(t *testing.T) {
 		})
 	}
 
-	if err := createDataset(ioutil.Discard, tc.ProjectID, location, datasetID); err != nil {
-		t.Skipf("Unable to create test dataset: %v", err)
-		return
+	{
```
Comment (enocom) [inline healthcare/dicom_test.go:62]: Ditto above.

File: healthcare/fhir_test.go [hunk 2]
```diff
@@ -50,9 +50,13 @@ func TestFHIRStore(t *testing.T) {
 		})
 	}
 
-	if err := createDataset(ioutil.Discard, tc.ProjectID, location, datasetID); err != nil {
-		t.Skipf("Unable to create test dataset: %v", err)
-		return
+	{
```
Comment (enocom) [inline healthcare/fhir_test.go:53]: And here.

File: healthcare/hl7v2_test.go [hunk 3]
```diff
@@ -44,11 +45,14 @@ func TestHL7V2Store(t *testing.T) {
 		})
 	}
 
-	testutil.Retry(t, 10, time.Second, func(r *testutil.R) {
-		if err := createDataset(ioutil.Discard, tc.ProjectID, location, datasetID); err != nil {
-			r.Errorf("Unable to create test dataset: %v", err)
+	{
```
Comment (enocom) [inline healthcare/hl7v2_test.go:48]: And here.

**Review by enocom — CHANGES_REQUESTED** [review]

File: healthcare/dataset_test.go [hunk 4]
```diff
@@ -50,9 +51,15 @@ func TestDataset(t *testing.T) {
 		})
 	}
 
-	if err := createDataset(buf, tc.ProjectID, location, datasetID); err != nil {
-		t.Fatalf("createDataset got err: %v", err)
+	{
```
[the remark this answers is not in the document]
Comment (tritone) [reply healthcare/dataset_test.go:54]: +1, better to use a named context.

File: healthcare/dataset_test.go [hunk 5]
```diff
@@ -50,9 +51,15 @@ func TestDataset(t *testing.T) {
...
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+		if err := createDataset(ctx, buf, tc.ProjectID, location, datasetID); err != nil {
+			cancel()
```
Comment (tritone) [inline healthcare/dataset_test.go:57]: I don't believe cancelling explicitly is necessary when you use a timeout... @enocom can you sanity check me on that?

File: healthcare/dataset_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (enocom) [reply healthcare/dataset_test.go:54]: A new variable name is more conventional in my opinion and reads a little better as not everyone knows this trick.

File: healthcare/dataset_create.go [hunk 6]
```diff
@@ -19,14 +19,13 @@ import (
...
 
 	healthcare "google.golang.org/api/healthcare/v1"
 )
 
 // createDataset creates a dataset.
-func createDataset(w io.Writer, projectID, location, datasetID string) error {
-	ctx := context.Background()
-
+func createDataset(ctx context.Context, w io.Writer, projectID, location, datasetID string) error {
 	healthcareService, err := healthcare.NewService(ctx)
```
Comment (tritone) [inline healthcare/dataset_create.go:29]: IMO it's better to create the context in the sample and set a timeout explicitly. This also is required by https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments Users will have to do this themselves in any case.

See example here: https://github.com/GoogleCloudPlatform/golang-samples/blob/HEAD/storage/buckets/create_bucket_class_location.go#L39

**Review by tritone — APPROVED** [review]

File: healthcare/dataset_create.go (hunk 6 above)
Comment (nikklassen) [reply healthcare/dataset_create.go:29]: thanks for pointing that out, I'll prefer following the style guide

**Comment by enocom** [comment]
I'll leave this to @tritone to approve.

**Pull request merged**